src/netstandard/WampSharp/WAMP2/V2/Api/CalleeProxy/CalleeProxyInterceptor.cs

```diff
@@ -27,7 +28,8 @@ public virtual CallOptions GetCallOptions(MethodInfo method)
         {
             CallOptions result = new CallOptions(mCallOptions);
 
-            if (method.IsDefined(typeof (WampProgressiveResultProcedureAttribute)))
+            if (method.IsDefined(typeof (WampProgressiveResultProcedureAttribute)) ||
```

Comment:
I think that the `WampProgressiveResultProcedureAttribute` should not be optional: users should explicitly state that they are invoking a progressive method and we should not implicitly determine this by the method signature (similar to the way we don't implicitly determine a method is a progressive method just by the fact that the method receives an IProgress and returns a Task).

Reply:
Ok I'll fix.

Out of curiosity, why make this explicit though? Isn't it obvious the caller expects progressive values when using `IObservable` return values or (even more so) when using the `IProgress` interface?
Or to put it in other words, would there ever be a situation where a user implements one of these interfaces and doesn't use the progress option, other than the situation where this is a mistake?

Reply:
The problem is that people make mistakes. Suppose that your method returns `IObservable` and everything work as expected. Then you refactor your code and change the return type to be something implementing `IObservable`. The code still runs but the behavior is significantly different. The framework doesn't warn you about anything. Placing the attribute allows the framework to validate your method signature and avoiding these time consuming issues. We should therefore also change the code in [MethodInfoValidation](https://github.com/Code-Sharp/WampSharp/blob/82a672e38cdf9489968d66d8b18a926cfd198ff7/src/netstandard/WampSharp/WAMP2/V2/Api/CalleeProxy/CalleeProxyInterceptorFactory.cs#L42) so that it first checks if the method has the `WampProgressiveResultProcedureAttribute` and only then if it does, checks whether the method returns `IObservable` or whether it returns a `Task`, and only then call `MethodInfoValidation.ValidateProgressiveMethod`. We should notify the user otherwise.